indico/modules/designer/client/js/index.js

```diff
@@ -424,9 +430,54 @@ import {$T} from 'indico/utils/i18n';
           selectedItem.y = unzoom(y > 0 ? y : 0);
         }
       },
+      back: function() {
+        if (div) {
+          var zIndex = getZIndex(selectedItem, false);
+          div.css('zIndex', zIndex);
+          selectedItem.zIndex = zIndex;
+        }
+      },
+      front: function() {
+        if (div) {
+          var zIndex = getZIndex(selectedItem, true);
+          div.css('zIndex', zIndex);
+          selectedItem.zIndex = zIndex;
+        }
+      },
     }[direction]());
   }
 
+function itemsOverlap(item1, item2) {
+    const {x: left1, y: top1} = item1;
+    const {x: left2, y: top2} = item2;
+
+    const right1 = left1 + item1.width;
+    const bottom1 = top1 + (item1.height || 22);
+    const right2 = left2 + item2.width;
+    const bottom2 = top2 + (item2.height || 22);
+
+    if (left1 >= right2 || left2 >= right1) {
+      return false;
+    }
+    return !(top1 >= bottom2 || top2 >= bottom1);
+  }
+
+  function getZIndex(selectedItem, moveToFront) {
+    let zIndex = selectedItem.zIndex;
+    Object.values(items)
+      .filter(item => selectedItem.id !== item.id)
+      .forEach(item => {
```

Comment:
Not a very large performance penalty, but it would be more efficient to do the ID check inside the `forEach(...)` closure.